src/scripts/qrHandler.js

```diff
@@ -1,14 +1,26 @@
-import QRCode from "davidshimjs-qrcodejs";
+import QRCodeStyling from "qr-code-styling";
 
-const createQRCodeFor = (identifier, dimension) => {
-    return new QRCode(identifier, {
+const createQRCodeFor = (dimension) => {
+    const qrCode = new QRCodeStyling({
         width: dimension,
         height: dimension,
-        colorDark: "#000000",
-        colorLight: "#ffffff",
-        correctLevel: QRCode.CorrectLevel.L,
+        type: "canvas",
+        data: cid,
+        image: "https://raw.githubusercontent.com/p2plabsxyz/dscan/main/public/logo.png", // default
+        dotsOptions: {
+            color: "#000000", // default
+            type: "rounded"
+        },
+        backgroundOptions: {
+            colorLight: "#ffffff", // default
+        },
+        imageOptions: {
+            crossOrigin: "anonymous",
+            margin: 2
+        }
     });
+    return qrCode;
 };
 
-export var QR_CODE_DOWNLOAD = createQRCodeFor("qrcode-download", 400);
-export var QR_CODE_DISPLAY = createQRCodeFor("qrcode", 150);
+export var QR_CODE_DOWNLOAD = createQRCodeFor(400);
+export var QR_CODE_DISPLAY = createQRCodeFor(150);
```

Comment:
i think the issue is with the library itself
i tried it over here -> https://qr-code-styling.com/

<img width="657" alt="p3" src="https://github.com/p2plabsxyz/dscan/assets/105343290/5194b65e-0c7f-4ce3-8eaf-9f66ba0e8c79">

Reply:
Yes, it's a problem with the library. But this might help:
- https://github.com/kozakdenys/qr-code-styling/issues/104
- https://github.com/kozakdenys/qr-code-styling/issues/61

I haven't found any open-source libraries for QR code styling. If you come across one, please let me know, and feel free to integrate it.

Reply:
have a look at these ones - https://www.qrcode-monkey.com/qr-code-api-with-logo/

Reply:
kindly take a look at these - https://www.qrcode-monkey.com/qr-code-api-with-logo/

Reply:
I think playing with these values might do the trick:

```javascript
    const qrCode = new QRCodeStyling({
        width: 150,
        height: 150,
```

```javascript
export var QR_CODE_DISPLAY = createQRCodeFor(160);
```

Nvm, I'll look into this. Can you finish the rest of the requested changes?

Reply:
i have corrected rest of the requested changes, making PR soon